### PR TITLE
Add initial navigation tests

### PR DESCRIPTION
**Pull request by Scony**
Adds few initial test cases in the navigation area

### CODE REVIEW

File: tests/scene/test_navigation_agent_2d.h [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+		SceneTree::get_singleton()->get_root()->add_child(node_2d);
+
+		NavigationAgent2D *agent_node = memnew(NavigationAgent2D);
+
+		// agent should not be attached to any map when outside of tree
+		CHECK_FALSE(agent_node->get_navigation_map().is_valid());
+
+		SUBCASE("Agent should attach to default map when it enters the tree") {
+			node_2d->add_child(agent_node);
+			CHECK(agent_node->get_navigation_map().is_valid());
```
Comment (DarkKilauea) [inline tests/scene/test_navigation_agent_2d.h:60]: Should probably also check that the attached map has the same RID as the default map on the world.

File: tests/scene/test_navigation_agent_3d.h [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+		SceneTree::get_singleton()->get_root()->add_child(node_3d);
+
+		NavigationAgent3D *agent_node = memnew(NavigationAgent3D);
+
+		// agent should not be attached to any map when outside of tree
+		CHECK_FALSE(agent_node->get_navigation_map().is_valid());
+
+		SUBCASE("Agent should attach to default map when it enters the tree") {
+			node_3d->add_child(agent_node);
+			CHECK(agent_node->get_navigation_map().is_valid());
```
Comment (DarkKilauea) [inline tests/scene/test_navigation_agent_3d.h:60]: Same issue as the 3D agent, should check that the map of the agent matches the default map of the world.

File: tests/servers/test_navigation_server_2d.h [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+#include "servers/navigation_server_2d.h"
+
+#include "tests/test_macros.h"
+
+namespace TestNavigationServer2D {
+TEST_SUITE("[Navigation]") {
+	TEST_CASE("[NavigationServer2D] Server should be empty when initialized") {
+		NavigationServer2D *navigation_server = NavigationServer2D::get_singleton();
+		CHECK_EQ(navigation_server->get_maps().size(), 0);
```
Comment (DarkKilauea) [inline tests/servers/test_navigation_server_2d.h:42]: Also check that regions, links, and agents are all empty?

File: tests/servers/test_navigation_server_3d.h [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+#include "servers/navigation_server_3d.h"
+
+#include "tests/test_macros.h"
+
+namespace TestNavigationServer3D {
+TEST_SUITE("[Navigation]") {
+	TEST_CASE("[NavigationServer3D] Server should be empty when initialized") {
+		NavigationServer3D *navigation_server = NavigationServer3D::get_singleton();
+		CHECK_EQ(navigation_server->get_maps().size(), 0);
```
Comment (DarkKilauea) [inline tests/servers/test_navigation_server_3d.h:42]: Also check regions, links, and agents?

File: tests/test_main.cpp [hunk 5]
```diff
@@ -248,6 +253,12 @@ struct GodotTestCaseListener : public doctest::IReporter {
 			audio_server->init();
 			return;
 		}
+
+		if (suite_name.find("[Navigation]") != -1 && navigation_server_2d == nullptr && navigation_server_3d == nullptr) {
```
Comment (DarkKilauea) [inline tests/test_main.cpp:257]: These servers are already created up on line 231 and 232, I don't believe this initialization is actually doing anything.

**Review by DarkKilauea — CHANGES_REQUESTED** [review]
Looks pretty good, left a few comments.

File: tests/test_main.cpp (hunk 5 above)
Comment (Scony) [reply tests/test_main.cpp:257]: lines 231 and 232 are called for tests with `[SceneTree]` substring which is not the case in the case of server-only tests from `[Navigation]` test suite

File: tests/servers/test_navigation_server_2d.h (hunk 3 above)
Comment (Scony) [reply tests/servers/test_navigation_server_2d.h:42]: There is no public API for that. That would be only doable in the context of the existing map if we'd have one.

File: tests/servers/test_navigation_server_3d.h (hunk 4 above)
Comment (Scony) [reply tests/servers/test_navigation_server_3d.h:42]: There is no public API for that. That would be only doable in the context of the existing map if we'd have one.

File: tests/scene/test_navigation_agent_2d.h (hunk 1 above)
Comment (Scony) [reply tests/scene/test_navigation_agent_2d.h:61]: done

File: tests/scene/test_navigation_agent_3d.h (hunk 2 above)
Comment (Scony) [reply tests/scene/test_navigation_agent_3d.h:60]: done

File: tests/servers/test_navigation_server_2d.h (hunk 3 above)
Comment (DarkKilauea) [reply tests/servers/test_navigation_server_2d.h:42]: ah, nevermind, those are map specific.

**Review by DarkKilauea — APPROVED** [review]

**Comment by Scony** [comment]
@akien-mga since these are just tests, I think we're ready to merge this.

**Review by akien-mga — APPROVED** [review]

**Pull request merged**